src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+
+import { NodeEvent, TreeModel } from 'ng2-tree';
+
+import { DashboardService } from '../../../shared/api/dashboard.service';
+
+@Component({
+  selector: 'cd-crushmap',
+  templateUrl: './crushmap.component.html',
+  styleUrls: ['./crushmap.component.scss']
+})
+export class CrushmapComponent implements OnInit {
+  panelTitle: string;
+  treeData: TreeModel;
+  crushMetaDatas: any[];
+  metadata: any;
+  metadataKeyMap: { [key: number]: number } = {};
+
+  constructor(private dashboardService: DashboardService) {
+    this.panelTitle = 'Cluster Crush Map';
+  }
+
+  ngOnInit() {
+    this.dashboardService.getHealth().subscribe((data: any) => {
+      this.crushMetaDatas = data.osd_map.tree.nodes || [];
+      this.treeData = this._abstractTreeData(data);
+    });
+  }
+
+  _abstractTreeData(data: any): TreeModel {
+    const cephNodes = data.osd_map.tree.nodes || [];
+    const nodesLength = cephNodes.length;
+    const tmpMap: { [key: number]: any } = {};
+
+    if (0 === nodesLength) {
+      return { value: 'No nodes!' };
+    }
+
+    for (let i = nodesLength - 1; i > -1; i--) {
+      const id = cephNodes[i].id;
+      this.metadataKeyMap[id] = i;
+      let value: string = cephNodes[i].name + ' (' + cephNodes[i].type + ')';
+      const children: any[] = [];
+
+      if (cephNodes[i].crush_weight) {
+        value += '--' + cephNodes[i].crush_weight.toString();
+      }
+
+      if (cephNodes[i].children) {
+        const len = cephNodes[i].children.length;
+        for (let j = 0; j < len; j++) {
+          children.push(tmpMap[cephNodes[i].children[j]]);
+        }
+        tmpMap[id] = { value, id, children };
+      } else {
+        tmpMap[id] = { value, id };
+      }
+
+      if (0 === i) {
+        return tmpMap[id];
```

Comment:
Please set the tree static to remove the context menu.
```suggestion
        tmpMap[id].settings = { static: true };
        return tmpMap[id];
```